### ISSUE TITLE
Can NOT speed up with DDP

### ISSUE THREAD

**Issue by poor1017** [speechbrain/speechbrain#1517]
Hi

I can not speed up training with DDP for `recipes/WHAMandWHAMR/enhancement/train.py`. My DDP command line is:

```
python3 -m torch.distributed.launch --nproc_per_node=4 --master_port 6666 train.py hparams/sepformer-whamr-16k-DM.yaml --data_folder /workspace/cpfs-data/Workstations/speechbrain/recipes/WHAMandWHAMR/data/whamr --base_folder_dm /workspace/cpfs-data/Data/wham/wsj0/si_tr_s/ --rir_path rir_wavs --distributed_launch --distributed_backend='nccl'
```

Compare to:

```
python3 train.py hparams/sepformer-whamr-16k-DM.yaml --data_folder /workspace/cpfs-data/Workstations/speechbrain/recipes/WHAMandWHAMR/data/whamr --base_folder_dm /workspace/cpfs-data/Data/wham/wsj0/si_tr_s/ --rir_path rir_wavs
```

DDP mode cost almost the same time.

Could you help to check this issue?

Thank you!

**Comment by poor1017** [comment]
Hi @Samuele Cornell, @Cem Subakan
I found the reason that makes DDP do not work:
[https://github.com/speechbrain/speechbrain/blob/develop/recipes/WHAMandWHAMR/enhancement/dynamic_mixing.py#L216](url)

Can you tell me the reason you wrap `train_data` with `torch.utils.data.DataLoader`?

Thank you

**Comment by popcornell** [comment]
Hi, your tag did not work so I ve missed this issue sorry, thanks to @anautsch that pinged me. 
Does it happens only with Dynamic Mixing ? Or also with the plain recipe ?

**Comment by ycemsubakan** [comment]
Also, @poor1017 , do you see the same behavior with WSJ0Mix recipe also?

**Comment by poor1017** [comment]
@popcornell Only with Dynamic Mixing, because with DM, you use torch.utils.data.DataLoader to cover sb.dataio.dataset.DynamicItemDataset
@ycemsubakan I have not tried WSJ0Mix.

**Comment by mravanelli** [comment]
Samuele, could you please take a look at it?

**Comment by Adel-Moumen** [comment]
Hello, I believe that this issue is related to #1802 and should be solved in #2010.

**Comment by Adel-Moumen** [comment]
Closing this issue as #2010 is merged in develop.
